roles/pulp/defaults/main.yml

```diff
@@ -0,0 +1,11 @@
+---
+pulp_image: pulp/pulp:stable
+pulp_ports:
+  - "8080:80"
+pulp_volumes:
+  - pulp_settings.volume:/etc/pulp:Z
```

Comment:
Since you're providing the setting through an environment variable and don't actually write out a settings file, I think you can drop this volume. Not entirely sure if Pulp actually creates files at runtime though.